client/webserver/site/src/js/order.ts

```diff
@@ -310,10 +368,13 @@ export default class OrderPage extends BasePage {
 
     Doc.setVis(!m.isCancel && (makerSwapCoin(m) || !m.revoked), tmpl.makerSwap)
     Doc.setVis(!m.isCancel && (takerSwapCoin(m) || !m.revoked), tmpl.takerSwap)
-    Doc.setVis(!m.isCancel && (makerRedeemCoin(m) || !m.revoked), tmpl.makerRedeem)
+    // When revoked, there is no need to show maker redeem if we are maker (since
+    // it won't happen), but if we are taker, maker redeem might still show up
+    // if maker "doesn't play by server rules" (so we have to account for that).
+    Doc.setVis(!m.isCancel && (makerRedeemCoin(m) || !m.revoked || (m.revoked && m.side === OrderUtil.Taker)), tmpl.makerRedeem)
```

Comment:
I wonder if this part (it seems, this code makes this assumption) I documented here: `When revoked, there is no need to show maker redeem if we are maker (since it won't happen)` is always **true** ?

Could the following happen ? Maker `dexc` publishes maker redeem transaction roughly at the moment of match revocation, such that redeem transaction will be mined **after** match is already revoked, resulting in:
- maker actually redeemed
- but (as Maker) we don't show it in UI

that would be bad ^, it seems to me ...

Reply:
Or maybe (for Maker) when Maker sent redeem transaction and it is waiting in mempool - `makerRedeemCoin(m)` will already have it for display ?